source/fundamentals/serialization/polymorphic-objects.txt

```diff
@@ -156,7 +156,19 @@ you must explicitly list each class you're looking for:
 .. code-block:: csharp
    :copyable: true
    
-   var query = coll.Aggregate().Match(a => a is Cat || a is Lion || a is Tiger);
+   var query = coll.AsQueryable().Where(
+       item => item.GetType() == typeof(Cat) ||
+       item.GetType() == typeof(Lion) ||
+       item.GetType() == typeof(Tiger));
+
+.. note:: OfType<T>() and the is Operator
+
+   When checking the type of a scalar discriminator, use the ``Where`` syntax shown in
+   the preceding code example. If you try to use the ``Aggregate().OfType<T>()`` method,
+   or if you pass an expression containing the ``is`` operator to the
+   ``Aggregate().Match()`` method, the driver throws an exception.
+
+.. _csharp-discriminator-hierarchical:
```

Comment:
This limitation will go away when https://jira.mongodb.org/browse/CSHARP-5356 is merged.

Reply:
We'll publish this now and can update in 3.2 or whenever the fix is implemented.